### ISSUE TITLE
Support glob-watching

### ISSUE THREAD

**Issue by jimporter** [jimporter/bfg9000#1]
To my knowledge, there's no existing meta-build system that supports globbing source files *and* regenerating the build files if the result of the glob changes. For instance, while CMake supports globbing and automatic build file regeneration, the two can't be used together [[1](http://www.cmake.org/pipermail/cmake/2009-January/026486.html)].

Obviously, the hard part here is doing this fast: incremental builds should *only* regenerate the build files if a relevant file was added or deleted. Here's a potential solution:

* Provide built-in globbing/find functions that remember their search patterns
* When initially generating a build file, create a `.bfg_watch` file that lists these patterns and their results
* When performing an incremental build, open the `.bfg_watch` file, and go through each of the glob patterns to see if their results changed
  * Lots of optimization potential here: for instance, check the mtime of directories to see if their file list has changed at all; otherwise, it can be skipped
* If so, regenerate the build file

**Comment by jimporter** [comment]
Here are some tricks to get the build files to handle this (`check-for-updated-globs` will touch `.bfg_watch` if the globs have changed):

Make:
```
__UNUSED__ := $(shell check-for-updated-globs)

Makefile: .bfg_watch
	regenerate-makefile
```

Ninja:
```
rule configure
  command = regenerate-makefile
  generator = 1

rule update
  command = check-for-updated-globs
  restat = 1

build .dirty: phony
build .bfg_watch: update | .dirty

build build.ninja: configure | .bfg_watch
```

**Comment by jimporter** [comment]
Fixed in e0ed852db01fc4dead4c7f7a8a7f09aec5fb98e8
